packages/palette/src/elements/Carousel/utils.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Sums all values, building from the previous
+ * @param xs list of values to compound
+ */
+export const compound = (xs: number[]): number[] => {
+  return xs.reduce((p, n) => p.concat((+p.slice(-1) || 0) + n), [])
+}
+
+/**
+ * Build an array of offsets
+ * @param viewport window width
+ * @param values array of cell widths
+ */
+export const chunk = (viewport: number, values: number[]) => {
+  return values.reduce(
+    (acc, value) => {
+      const prev = acc[acc.length - 1]
+      if (prev + value <= viewport) {
+        acc[acc.length - 1] += value
+        return acc
+      }
+      return [...acc, value]
+    },
+    [0]
+  )
+}
+
+/** Align last page to the left or right */
+export type Alignment = "left" | "right"
+
+/**
+ * Creates an array of offsets based on the given viewport
+ * @param viewport window width
+ * @param values array of cell widths
+ * @param alignment align last page to the left or right
+ */
+export const paginate = (
```

Comment:
This makes sense to keep in the carousel folder, but split into its own file called `paginate`.